Studies/source/GenStudy.cxx

```diff
@@ -529,15 +481,46 @@ class GenStudy {
                     }
                 }
             }
+            if(b_jet_matches_total.size() == 2 && tau_matches_total.size() ==2) {
+                anaData.n("2_matches_for_all").Fill(1);
+                for (size_t reco_tau_index = 0; reco_tau_index < event.lep_p4.size(); reco_tau_index++) {
+                    int matched_gen_tau = -1;
+                    for (int gen_tau_index = 0; gen_tau_index < HH_Gen_Event.tau.size(); gen_tau_index++) {
+                        if(tau_matches.at(gen_tau_index).count(reco_tau_index)) {
+                            matched_gen_tau = gen_tau_index;
+                            break;
+                        }
+                    }
+                    (*new_tuple)().lep_genTauIndex.push_back(matched_gen_tau);
+                    new_tuple->Fill();
+                }
+                for (size_t reco_b_index = 0; reco_b_index < event.jets_p4.size(); reco_b_index++) {
+                    int matched_gen_b = -1;
+                    for (int gen_b_index = 0; gen_b_index < HH_Gen_Event.b_jets.size(); gen_b_index++) {
+                        if(b_matches.at(gen_b_index).count(reco_b_index)) {
+                            matched_gen_b = gen_b_index;
+                            break;
+                        }
+                    }
+                    (*new_tuple)().jets_genJetIndex.push_back(matched_gen_b);
+                    new_tuple->Fill();
+                }
+                (*new_tuple)() = event;
```

Comment:
this should be done before filling lep_genTauIndex and jets_genJetIndex, otherwise, the results are erased.